app/controllers/proposals_controller.rb

```diff
@@ -28,4 +42,17 @@ def auth_errors(exception)
     redirect_to proposals_path,
       alert: "You are not allowed to see that proposal"
   end
+
+  # Used in the query template to provide a span of time in the header
+  def datespan_header
+    if @start_date && @end_date
+      # month span
+      if @start_date.mday == 1 && @end_date == @start_date + 1.month
+        month_name = I18n.t('date.abbr_month_names')[@start_date.month]
+        "(#{month_name} #{@start_date.year})"
+      else
+        "(#{@start_date.iso8601} - #{@end_date.iso8601})"
+      end
+    end
+  end
```

Comment:
Since it's only used in the view, can we put it in `app/helpers/proposals_helper.rb`?